gcsio/src/main/java/com/google/cloud/hadoop/gcsio/TraceLoggingHttpRequestInitializer.java

```diff
@@ -0,0 +1,69 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpResponse;
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.Payload.JsonPayload;
+import com.google.cloud.logging.Severity;
+import com.google.common.base.Stopwatch;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+class TraceLoggingHttpRequestInitializer implements HttpRequestInitializer {
+
+  private final Map<HttpRequest, Stopwatch> requestTracker = new ConcurrentHashMap<>();
+  private final Logging logging;
+
+  public TraceLoggingHttpRequestInitializer(Logging logging) {
+    this.logging = logging;
+  }
+
+  @Override
+  public void initialize(HttpRequest request) throws IOException {
+    request.setInterceptor(
+        httpRequest -> requestTracker.put(httpRequest, Stopwatch.createStarted()));
+    request.setResponseInterceptor(this::log);
+  }
+
+  private void log(HttpResponse httpResponse) {
```

Comment:
Google API client library has already very low-level trace logging of executed request, did we explore if they are suitable for debugging purposes?